src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/clusterinteraction/ClusterDetailFragment.kt

```diff
@@ -49,45 +52,120 @@ class ClusterDetailFragment : Fragment() {
     get() = ChipClient.getDeviceController(requireContext())
 
   private val scope = CoroutineScope(Dispatchers.Main + Job())
-  private lateinit var clusterMap: Map<String, ClusterInfo>
-  private lateinit var selectedClusterInfo: ClusterInfo
-  private lateinit var selectedCluster: ChipClusters.BaseChipCluster
-  private lateinit var selectedCommandCallback: DelegatedClusterCallback
-  private lateinit var selectedInteractionInfo: InteractionInfo
-  private var devicePtr = 0L
-  private var endpointId = 0
 
   override fun onCreateView(
     inflater: LayoutInflater,
     container: ViewGroup?,
     savedInstanceState: Bundle?
   ): View {
-    clusterMap = ClusterInfoMapping().clusterMap
-    devicePtr = checkNotNull(requireArguments().getLong(DEVICE_PTR_KEY))
     endpointId = checkNotNull(requireArguments().getInt(ENDPOINT_ID_KEY))
+    historyCommand = if (requireArguments().getSerializable(HISTORY_COMMAND) == null) {
+      HistoryCommand("", "", mutableListOf(), emptyMap<CommandResponseInfo, Any>(), "")
```

Comment:
historyCommand isn't used at all if HISTORY_COMMAND is null, so why not just make historyCommand nullable, instead of setting a placeholder `HistoryCommand`, which is hacky?